Upgrade Plug to require Elixir ~> 1.2

whatyouhide opened the pull request.

josevalim commented:
The rand module is available only from Erlang 18, which means we would need
to depend on Elixir v1.2 (which is fine by me).

On Thursday, June 30, 2016, Andrea Leopardi <notifications@github.com>
wrote:

> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/elixir-lang/plug/pull/416
> Commit Summary
>
>    - Silence stuff that was being logged by Plug.DebuggerTest
>    - Use the :rand module instead of :random
>
> File Changes
>
>    - *M* test/plug/conn_test.exs
>    <https://github.com/elixir-lang/plug/pull/416/files#diff-0> (8)
>    - *M* test/plug/debugger_test.exs
>    <https://github.com/elixir-lang/plug/pull/416/files#diff-1> (30)
>
> Patch Links:
>
>    - https://github.com/elixir-lang/plug/pull/416.patch
>    - https://github.com/elixir-lang/plug/pull/416.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/elixir-lang/plug/pull/416>, or mute the thread
> <https://github.com/notifications/unsubscribe/AAAlblm-2YqBR0s-W7Ri_d4aRVKV8wUAks5qQ9e6gaJpZM4JCPg0>
> .
>


-- 


*José Valim*
www.plataformatec.com.br
Skype: jv.ptec
Founder and Director of R&D

josevalim commented:
Therefore it may be worth checking if there is any TODO or other
improvements we could do for v1.2. I think for example there are parts of
the query encoder that could use URI functions as they have been optimized
in Elixir since then. As well as possible usages of Erlang:now and other
otp_release checks.

-- 


*José Valim*
www.plataformatec.com.br
Skype: jv.ptec
Founder and Director of R&D

whatyouhide commented:
@josevalim good point, I'll see what we can fix if we depend on Erlang 18.

josevalim commented:
Hopefully we left some good amount of TODOs in the code. :)

On Thursday, June 30, 2016, Andrea Leopardi <notifications@github.com>
wrote:

> @josevalim <https://github.com/josevalim> good point, I'll see what we
> can fix if we depend on Erlang 18.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/elixir-lang/plug/pull/416#issuecomment-229693809>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/AAAlbqsk0Z3AalfQZtUNVpLlYMiEVib9ks5qQ9-qgaJpZM4JCPg0>
> .
>


-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*

whatyouhide commented:
@josevalim the only `TODO` I found was about using `System.monotonic_time/1` if available. I also fixed a bunch of `capture_log`-related stuff and updated `mix.exs` to require ~> 1.2 as well as updated the changelog. Do we need to do anything else?

whatyouhide commented:
Oh, and I update `.travis.yml` with the correct test matrix as well.

josevalim commented:
Ship it!

On Thursday, June 30, 2016, Andrea Leopardi <notifications@github.com>
wrote:

> Oh, and I update .travis.yml with the correct test matrix as well.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/elixir-lang/plug/pull/416#issuecomment-229704677>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/AAAlbora8ik8tvQ_GOw5EUaNeJFvw7Z0ks5qQ-gHgaJpZM4JCPg0>
> .
>


-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*
